### PR TITLE
Development/testing

### PR DESCRIPTION
**Pull request by domenico-villani**
- a shell script that'll make the Stevedore artifact zips

- a new Yamato YAML file for a "Publish Stevedore artifacts" Yamato job that automatically uploads stevedore artifacts after Roslyn is built, it uploads to testing for now

### CODE REVIEW

File: .yamato/publish_stevedore_artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: 'Publish 7z Stevedore artifacts'
+
+agent:
+  type: Unity::VM
+  image: cds-ops/cds-ubuntu-18.04-base:stable
+  flavor: b1.small
+dependencies:
+  - .yamato/build.yml
+commands:
+  - sudo apt-get install -y p7zip-full p7zip-rar
```
Comment (kwi-dk) [inline .yamato/publish_stevedore_artifacts.yml]: I see on Yamato that the image already has p7zip-full:
> p7zip-full is already the newest version (16.02+dfsg-6).

And you surely don't need RAR support? So you can skip this step entirely.

File: .yamato/test.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+name: 'Test Mono Upload'
```
Comment (kwi-dk) [inline .yamato/test.yml]: This entire file should presumably just be deleted, now you're done testing?

File: stevedore-build-artifacts.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+version_string="$1"
+artifact_path='producedbuilds/*'
+
+if [ ! "$version_string" ]; then
+    echo "missing arg: version_string"
+    exit 1
+fi
+
+sudo chmod -v +x Artifacts/Builds/Binaries/Linux/csc
```
Comment (kwi-dk) [inline stevedore-build-artifacts.sh]: I don't think you need `sudo`, the entire script should already be running as root? (And if not, you should still not need root to set the execute bit.)

File: stevedore-build-artifacts.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+7z a producedbuilds/roslyn-csc-mac.7z $PWD/Artifacts/Builds/Binaries/Mac/*
+7z a producedbuilds/roslyn-csc-win64.7z $PWD/Artifacts/Builds/Binaries/Windows/*
+7z a producedbuilds/roslyn-csc-net46.7z $PWD/Artifacts/Builds/Binaries/Net46/*
+
+for path in $artifact_path;do
+  artifact_hash="$(sha256sum "$path" | cut -f1 -d ' ')"
+  source_artifact_id="$(basename "$PWD/$path" .7z)/${version_string}_${artifact_hash}.7z"
+
+  echo "source artifact id: $source_artifact_id"
+done
```
Comment (kwi-dk) [inline stevedore-build-artifacts.sh]: This seems like a lot of work just to print out the artifact IDs. `StevedoreUpload.exe` already prints out the artifact IDs upon successful upload, e.g.:

    Artifact uploaded successfully:
        testing: roslyn-csc-linux/41f6ba9d99e7_6b3d1c740d0840e1eb1c58421ffe3ad5cf5c9e25c70e5ec78550bd42daa08dde.7z

Seems like you can remove all this, and then don't need to pass any arguments when running the script either.

File: .yamato/publish_stevedore_artifacts.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+name: 'Publish 7z Stevedore artifacts'
```
Comment (kwi-dk) [inline .yamato/publish_stevedore_artifacts.yml]: That should be "Publish roslyn-csc Stevedore artifacts" or something like that. :)

**Review by kwi-dk — COMMENTED** [review]
Added some comments.

File: .yamato/publish_stevedore_artifacts.yml (hunk 1 above)
Comment (domenico-villani) [reply .yamato/publish_stevedore_artifacts.yml]: done

File: .yamato/publish_roslyn-csc_stevedore_artifacts.yml [hunk 6]
```diff
@@ -7,8 +7,7 @@ agent:
 dependencies:
   - .yamato/build.yml
 commands:
-  - sudo apt-get install -y p7zip-full p7zip-rar
   # Script needs the version string of the source artifact (which must match StevedoreUpload args).
```
Comment (kwi-dk) [inline .yamato/publish_roslyn-csc_stevedore_artifacts.yml]: This comment is no longer valid.

**Review by kwi-dk — CHANGES_REQUESTED** [review]
One final comment, then we'll look at landing it.

File: stevedore-build-artifacts.sh [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+ls -l Artifacts/Builds/Binaries/Mac/csc
+chmod -v +x Artifacts/Builds/Binaries/Mac/VBCSCompiler
+ls -l Artifacts/Builds/Binaries/Mac/VBCSCompiler
+
+mkdir -p producedbuilds
+
+7z a producedbuilds/roslyn-csc-linux.7z $PWD/Artifacts/Builds/Binaries/Linux/*
+7z a producedbuilds/roslyn-csc-mac.7z $PWD/Artifacts/Builds/Binaries/Mac/*
+7z a producedbuilds/roslyn-csc-win64.7z $PWD/Artifacts/Builds/Binaries/Windows/*
+7z a producedbuilds/roslyn-csc-net46.7z $PWD/Artifacts/Builds/Binaries/Net46/*
```
Comment (kwi-dk) [inline stevedore-build-artifacts.sh:25]: It occurs to me that this is not adding a LICENSE file to the artifacts (per the artifact guidelines, "All artifacts must contain a LICENSE file in their root"... even if our current csc artifacts are possibly also missing this). You can simply use the existing License.txt from the Roslyn repo (though the filename must be changed to uppercase): it should be enough to simply run
```cp License.txt $PWD/Artifacts/Builds/Binaries/Linux/LICENSE.txt```
(etc., for each of the four artifacts), _before_ 7z.

**Review by kwi-dk — CHANGES_REQUESTED** [review]
One more.

**Comment by kwi-dk** [comment]
Superseded by #12.

**Pull request closed without merging**